references/classification/train.py

```diff
@@ -11,26 +11,38 @@
 
 import utils
 
+try:
+    from apex import amp, optimizers
+except ImportError:
+    print("Please install apex from https://www.github.com/nvidia/apex to enable mixed precision training.")
 
-def train_one_epoch(model, criterion, optimizer, data_loader, device, epoch, print_freq):
+
+def train_one_epoch(model, criterion, optimizer, data_loader, device, epoch, print_freq, apex=False):
     model.train()
     metric_logger = utils.MetricLogger(delimiter="  ")
     metric_logger.add_meter('lr', utils.SmoothedValue(window_size=1, fmt='{value}'))
+    metric_logger.add_meter('img/s', utils.SmoothedValue(window_size=10, fmt='{value}'))
+
     header = 'Epoch: [{}]'.format(epoch)
     for image, target in metric_logger.log_every(data_loader, print_freq, header):
         image, target = image.to(device), target.to(device)
         output = model(image)
         loss = criterion(output, target)
 
+        start_time = time.time()
```

Comment:
Is this necessary, given that we already log timing inside `metric_logger` for the full forward / backward / optimizer?

Reply:
maybe not, but I just want to have a simple and intuitive way to update the throughput instead of pulling it from `metric_logger`. Appreciate any suggestion though.

Reply:
The time for the total iteration gets printed every `print_freq` https://github.com/pytorch/vision/blob/47db963800649bdd609d619bbc5f08052fe5520a/references/classification/utils.py#L123

So I suppose this is enough and we can just remove this extra timing?

Reply:
I guess most people cannot easily relate to a time statistics (sec per batch) compared to a throughput statistics (imgs/sec), hence the reason for adding a throughput metric. If removing the added timing, do you have any suggestion to log the throughput?

Reply:
Let's keep this then, even though this is a bit redundant IMO.